### ISSUE TITLE
python 2.7 deprecated

### ISSUE THREAD

**Issue by max-endeavour** [CellProfiler/CellProfiler#4327]
**Describe the bug**
Hi got the error due python 2.X deprecation since 1st january2020
**To Reproduce**
1. I user the requirements.yaml provided
2. conda env create -f requirements.yaml
3.See error:

```
DEPRECATION: Python 2.7 will reach the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 won't be maintained after that date. A future version of pip will drop support for Python 2.7. More details about Python 2 support in pip, can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support
  Running command git clone -q https://github.com/CellProfiler/CellProfiler.git /tmp/pip-req-build-6zuMmr
ERROR: Package u'CellProfiler' requires a different Python: 2.7.18 not in '>=3.8, <4.0'

failed

CondaEnvException: Pip failed
```

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**
 - OS:Ubuntu 18.04
 - CellProfiler Version master branch
 - Downloaded from the website or installed from source? From sources
 - If the latter, how?

**Additional context**

**Comment by DavidStirling** [comment]
Where exactly did you get this `requirements.yaml` file? CellProfiler 4 runs on Python 3.8 and above, rather than Python 2. It's also generally best to install with `pip` rather than `conda`.

**Comment by max-endeavour** [comment]
Hi, went through the [conda install](https://github.com/CellProfiler/CellProfiler/wiki/Conda-Installation) as I use conda to manage my environments.
Looking back at it I used the cp4 requirements which worked.
Maybe consider removing the version 3 requirements and update the doc as it is not obvious that the new requirements needs to be used with the old procedure (why this file is not versioned with the rest of the code btw?)

**Comment by max-endeavour** [comment]
While able to install, this error raised ? any idea? do you want to open an other issue? 

```
cellprofiler
Gtk-Message: 16:22:23.588: Failed to load module "atk-bridge"
Gtk-Message: 16:22:23.589: Failed to load module "canberra-gtk-module"
16:22:23: Debug: Failed to connect to session manager: None of the authentication protocols specified are supported
16:22:24: Debug: Adding duplicate image handler for 'Windows bitmap file'
16:22:24: Debug: Adding duplicate animation handler for '1' type
16:22:24: Debug: Adding duplicate animation handler for '2' type
Traceback (most recent call last):
  File "/home/cnp-ds/anaconda3/envs/cp4/lib/python3.8/site-packages/cellprofiler/gui/app.py", line 69, in OnInit
    from .cpframe import CPFrame
  File "/home/cnp-ds/anaconda3/envs/cp4/lib/python3.8/site-packages/cellprofiler/gui/cpframe.py", line 25, in <module>
    from ._welcome_frame import WelcomeFrame
  File "/home/cnp-ds/anaconda3/envs/cp4/lib/python3.8/site-packages/cellprofiler/gui/_welcome_frame.py", line 7, in <module>
    import wx.html2
  File "/home/cnp-ds/anaconda3/envs/cp4/lib/python3.8/site-packages/wx/html2.py", line 15, in <module>
    from ._html2 import *
ImportError: libwebkitgtk-1.0.so.0: Ne peut ouvrir le fichier d'objet partagé: Aucun fichier ou dossier de ce type
OnInit returned false, exiting...
```
The terminal freezes there

**Comment by DavidStirling** [comment]
Glad you got it to install. In our experience Conda installations tend to create conflicts with the underlying Java environment required for CellProfiler to function properly, so our general advice is "just use pip". If users can help us to update the Conda instructions to be more reliable then that's always a good thing, but it's not been a priority for the development team.

Regarding your issue, we've seen this problem on some linux machines. It appears to be related to the wxPython wheels, which lack some necessary libraries. You can see [this page](https://github.com/CellProfiler/CellProfiler/wiki/Ubuntu-18.04) for more information.

**Comment by max-endeavour** [comment]
Hey,

Thank you for the feedback. I uninstalled cellprofiler from the cp4 environment and deactivated it. (Note that cellprofiler is installed using pip in the environment).
Once this done I installed cellprofiler using following the procedure in your link. (I used the wheel wxpython so skipped the procedure in the middle.)
The installation went ok, but when launching the cell profiler keen on having the same error.
